Mathlib/Topology/UniformSpace/Cauchy.lean

```diff
@@ -829,4 +829,42 @@ theorem secondCountable_of_separable [SeparableSpace α] : SecondCountableTopolo
     exact hUV (ball_subset_of_comp_subset (hk hxy) hUU' (hk hz))
 #align uniform_space.second_countable_of_separable UniformSpace.secondCountable_of_separable
 
+section DiscreteUniformity
+
+/-- A Cauchy filter in a discrete uniform space is contained in a principal filter-/
+theorem Cauchy_Discrete_le_principal {X : Type _} {uX : UniformSpace X}
+    (hX : uniformity X = 𝓟 idRel) {α : Filter X} (hα : Cauchy α) : ∃ x : X, α ≤ 𝓟 {x} := by
+  rcases hα with ⟨α_ne_bot, α_le⟩
+  rw [Filter.le_def] at α_le
+  specialize α_le idRel
+  simp only [le_def, hX, mem_principal, idRel_subset, mem_idRel, eq_self_iff_true, imp_true_iff,
+    forall_true_left, mem_prod_iff] at α_le
+  obtain ⟨_, ⟨hS, ⟨_, ⟨hT, H⟩⟩⟩⟩ := α_le
+  obtain ⟨x, hx⟩ :=
+    prod_subset_idRel_Eq_singleton_left (Filter.nonempty_of_mem hS) (Filter.nonempty_of_mem hT) H
+  use x
+  rwa [Filter.le_principal_iff, ← hx]
```

Comment:
I see many `Filter.` in this proof. Isn’t the namespace opened here?

Reply:
The two `Filter.nonempty_of_mem` on line 846 are needed because of some ambiguity, so I ended up not opening the namespace. Not it is opened and two `Flter.` are removed, I do not know if it is much better but am happy to keep it this way.